src/diagonal.works/b6/yaml.go

```diff
@@ -43,7 +49,18 @@ func unmarshalChoiceYAML(choices interface{}, unmarshal func(interface{}) error)
 	v := reflect.ValueOf(choices).Elem()
 	for i := 0; i < t.NumField(); i++ {
 		if f := v.Field(i); !f.IsNil() {
-			return f.Interface(), nil
+			// TO(mari): previously, when unmarshalling instances of expressions
+			// from YAML, we'd return a point to the unmarshalled expression
+			// (the `choices` struct has to contain pointers or slices, as we
+			// rely on only one value being filled in after unmarshalling, we
+			// have to be able to check whether the others a nil, which needs
+			// a type that can be nil). This meant that anything unmarshalled
+			// from yaml was a pointer. This bit below fixes that.
+			if f.Kind() != reflect.Slice {
+				return f.Elem().Interface(), nil
```

Comment:
wow i m glad i asked for help, finding / fixing on my own would ve taken a long time
ty !!

Reply:
and i remember seeing pointers in the choices / in the last cl / but i just didnt connect the two

Reply:
also the (pointer).equals(value) != (value).equals(pointer) totally escaped me